### ISSUE TITLE
Internal compiler when getters are used in module.exports when processing CommonJS modules

### ISSUE THREAD

**Issue by kommen** [google/closure-compiler#2510]
For example when consuming the [htmlparser2](https://github.com/fb55/htmlparser2) node module, the compiler crashes with:

```
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(SCRIPT): node_modules/htmlparser2/lib/index.js:1:0
var Parser = require("./Parser.js"),
  Parent: NULL
	at com.google.javascript.rhino.IR.mayBeExpression(IR.java:712)
	at com.google.javascript.rhino.IR.assign(IR.java:424)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.expandObjectLitAssignment(ProcessCommonJSModules.java:885)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.visitExport(ProcessCommonJSModules.java:792)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.visit(ProcessCommonJSModules.java:678)
	at com.google.javascript.jscomp.NodeTraversal.handleScript(NodeTraversal.java:581)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:604)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:305)
	at com.google.javascript.jscomp.NodeTraversal.traverseEs6(NodeTraversal.java:553)
	at com.google.javascript.jscomp.ProcessCommonJSModules.process(ProcessCommonJSModules.java:127)
	at com.google.javascript.jscomp.Compiler.processAMDAndCommonJSModules(Compiler.java:2107)
	at com.google.javascript.jscomp.Compiler.parseInputs(Compiler.java:1798)
	at com.google.javascript.jscomp.Compiler.parseForCompilationInternal(Compiler.java:1011)
	at com.google.javascript.jscomp.Compiler.access$300(Compiler.java:97)
	at com.google.javascript.jscomp.Compiler$7.call(Compiler.java:991)
	at com.google.javascript.jscomp.Compiler$7.call(Compiler.java:988)
	at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:93)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	... 21 more
```

The offending part seem to be the `get` definitions in the module exports:
https://github.com/fb55/htmlparser2/blob/2e6c0e0f3f79cc268efc59e73b25d917c07d8493/lib/index.js#L15

**Comment by kommen** [comment]
I've pushed a test case here: https://github.com/nextjournal/closure-compiler/commit/d5c931509c2245d8e655ecb188c31232175d5c11

**Comment by brad4d** [comment]
@kommen 
Thanks a lot for adding an example test case. That helps a lot.

The trouble is that the ProcessCommonJSModules pass runs really early, before any transpilation, but it hasn't been updated to understand setters and getters.

@concavelenz has suggested that the original intent when the pass was written was to only support "normal" properties.

I believe it should work fine to define the object first, then just do

```javascript
module.exports = export_object;
```

Short term we should add an explicit error for this case instead of throwing an NPE.
Since we want to focus on ES6 modules, I suspect we won't have time to devote to making this work for CommonJS modules.

**Comment by ChadKillingsworth** [comment]
Closed by 843a879
